src/dolphin/unwrap/_unwrap.py

```diff
@@ -272,12 +286,47 @@ def unwrap(
             output_filename=combined_mask_file,
         )
 
+    if run_goldstein:
+        suf = Path(unw_filename).suffix
+        if suf == ".tif":
+            driver = "GTiff"
+            opts = list(io.DEFAULT_TIFF_OPTIONS)
+        else:
+            driver = "ENVI"
+            opts = list(io.DEFAULT_ENVI_OPTIONS)
+
+        # If we're running Goldstein filtering, the intermediate
+        # filtered/unwrapped rasters are temporary rasters in the scratch dir.
+        filt_ifg_filename = (
+            Path(scratchdir or ".") / Path(ifg_filename).with_suffix(".filt" + suf).name
+        )
+        scratch_unw_filename = Path(unw_filename).with_suffix(".filt.unw" + suf)
+
+        ifg = io.load_gdal(ifg_filename)
+        logger.info(f"Goldstein filtering {ifg_filename} -> {filt_ifg_filename}")
+        filt_ifg = goldstein(ifg, alpha=alpha)
+        logger.info(f"Writing filtered output to {filt_ifg_filename}")
+        io.write_arr(
+            arr=filt_ifg,
+            output_name=filt_ifg_filename,
+            like_filename=ifg_filename,
+            driver=driver,
+            options=opts,
+        )
+        unwrapper_ifg_filename = filt_ifg_filename
+        unwrapper_unw_filename = scratch_unw_filename
+    else:
+        unwrapper_ifg_filename = Path(ifg_filename)
+        unwrapper_unw_filename = Path(unw_filename)
+
     if unwrap_method == UnwrapMethod.SNAPHU:
+        from ._snaphu_py import unwrap_snaphu_py
```

Comment:
Was it causing any problems to have this imported at the top of the file? Or was this a mistake to move the import?

Reply:
Oops, yeah I moved this here because I was testing in an environment without snaphu. It was very convenient to not have to install snaphu simply by moving the import to where it's needed, but I can definitely move this back if that's an antipattern.

Reply:
I suppose we can leave it for now if someone really want's to use another unwrapper that's not snaphu py, we're not currently putting `snaphu-py` in the environment file